manifests/templates/resourcegroup-manifest.yaml

```diff
@@ -231,6 +231,8 @@ spec:
             - ALL
       - args:
         - --config=/conf/otel-agent-config.yaml
+        - --feature-gates=-component.UseLocalHostAsDefaultHost
+        - --feature-gates=-pkg.translator.prometheus.NormalizeName
```

Comment:
What's pkg.translator.prometheus.NormalizeName for?

Reply:
This is for dropping the feature gate AKA disabling it, to revert to a simple fashion of name sanitizing, I realized metrics from reconilers and reconciler-manager are using this but not the resource-group-controller so adding it here. 

See https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/pkg/translator/prometheus/README.md

Reply:
Can you make this specific change in a followup PR as well? Seems like a bug fix. Need to explain what it actually changes and what was broken.

Reply:
Just removed the flag. Took a closer look - resource group controller only has gauge and distribution metrics, and the names are only composed with letters and underscores, so even with the normalizer on, no conversion is happening. Plus the test cases are not checking the r-g-c metrics, so no observations if there's a mismatch in name somehow.

We could add the flag just to be safe and align with other deployments.

Reply:
Feel free to add it in a new PR just for consistency.

Reply:
Ohhhh, I see, change was made a while back to actually enable this feature, but adding a `-` disabled it instead. https://github.com/GoogleContainerTools/kpt-config-sync/pull/922/commits/f3e2a5e1b371855afa83f89f009e2befd988bcdd

To re-enable it we could just remove the feature-gate flag as the feature is default on. But the trim & append will start to work and e2e test would fail (just ran a quick test).

From my reading of the doc it'd be better to have the normalizer running to convert the names for the favor of the Prometheus exporter/importer, which would break our e2e by a bit as it explicitly looks for the metric names defined and registered upon creation.

I lean keeping it as is and ignoring the naming, since any user with Prometheus operator would probably already hooked with the current naming.

Or we have a thorough & proper enablement of the feature.

WDYT @karlkfi